src/H5CX.c

```diff
@@ -3702,7 +3751,42 @@ H5CX_set_no_selection_io_cause(uint32_t no_selection_io_cause)
     } /* end if */
 
     FUNC_LEAVE_NOAPI_VOID
-} /* end H5CX_set_no_selectiion_io_cause() */
+} /* end H5CX_set_no_selection_io_cause() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5CX_set_actual_selecction_io_mode
+ *
+ * Purpose:     Sets the actual selection I/O mode for the current API
+ *              call context.
+ *
+ * Return:      <none>
+ *
+ * Programmer:  Vailin Choi
+ *              April 15, 2023
+ *
+ *-------------------------------------------------------------------------
+ */
+void
+H5CX_set_actual_selection_io_mode(uint32_t actual_selection_io_mode)
+{
+    H5CX_node_t **head = NULL; /* Pointer to head of API context list */
+
+    FUNC_ENTER_NOAPI_NOINIT_NOERR
+
+    /* Sanity checks */
+    head = H5CX_get_my_context(); /* Get the pointer to the head of the API context, for this thread */
+    HDassert(head && *head);
+    HDassert((*head)->ctx.dxpl_id != H5P_DEFAULT);
+
+    /* If we're using the default DXPL, don't modify it */
+    if ((*head)->ctx.dxpl_id != H5P_DATASET_XFER_DEFAULT) {
+        /* Cache the value for later, marking it to set in DXPL when context popped */
+        (*head)->ctx.actual_selection_io_mode     = actual_selection_io_mode;
+        (*head)->ctx.actual_selection_io_mode_set = TRUE;
+    } /* end if */
```

Comment:
/* end if */ comments are noise, especially when the if block is just a few lines long